uui-components/src/table/columnsConfigurationModal/columnsConfigurationUtils.ts

```diff
@@ -49,23 +49,30 @@ function isSubstring(s: React.ReactNode, sub: string) {
     }
     return false;
 }
-export function isColumnFilteredOut(c: DataColumnProps, filter?: string) {
-    const caption = c.caption;
+
+export function isColumnFilteredOut(column: any, searchFields: string[], filter?: string) {
+    const caption = column.caption;
     const hasCaption = !isEmptyCaption(caption);
     const hasFilter = !isEmptyString(filter);
+    const isNotSearchValue = hasFilter && !searchFields.some((searchField) => isSubstring(searchField, filter));
+    return hasCaption ? isNotSearchValue : true;
+}
 
-    return hasCaption ? hasFilter && !isSubstring(caption, filter) : true;
+interface IGroupAndFilterSortedColumnsProps<TItem, TId, TFilter> {
+    sortedColumns: ColumnsConfigurationRowProps[];
+    searchValue: string;
+    getSearchFields?: (column: DataColumnProps<TItem, TId, TFilter>) => string[];
 }
 
-export function groupAndFilterSortedColumns(sortedColumns: ColumnsConfigurationRowProps[], searchValue: string): GroupedColumnsType {
+export function groupAndFilterSortedColumns<TItem, TId, TFilter>(props: IGroupAndFilterSortedColumnsProps<TItem, TId, TFilter>): GroupedColumnsType {
     const accUnsorted = {
         displayedPinned: [],
         displayedUnpinned: [],
         hidden: [],
     } as GroupedColumnsType;
 
-    return sortedColumns.reduce((acc, i) => {
-        if (!isColumnFilteredOut(i, searchValue)) {
+    return props.sortedColumns.reduce((acc, i) => {
+        if (!isColumnFilteredOut(i, props?.getSearchFields ? props.getSearchFields(i) : [i.caption as string], props.searchValue)) {
```

Comment:
lets move this default inside UseColumnsConfiguration

Reply:
done